spring-ai-core/src/main/java/org/springframework/ai/chat/client/observation/DefaultChatClientObservationConvention.java

```diff
@@ -0,0 +1,144 @@
+		.of(ChatClientObservationDocumentation.HighCardinalityKeyNames.CHAT_CLIENT_ADVISORS, KeyValue.NONE_VALUE);
+
+	private static final KeyValue CHAT_CLIENT_ADVISOR_PARAM_NONE = KeyValue
+		.of(ChatClientObservationDocumentation.HighCardinalityKeyNames.CHAT_CLIENT_ADVISOR_PARAMS, KeyValue.NONE_VALUE);
+
+	private final String name;
+
+	public DefaultChatClientObservationConvention() {
+		this(DEFAULT_NAME);
+	}
+
+	public DefaultChatClientObservationConvention(String name) {
+		this.name = name;
+	}
+
+	@Override
+	public String getName() {
+		return this.name;
+	}
+
+	@Override
+	@Nullable
+	public String getContextualName(ChatClientObservationContext context) {
+		return "%s %s".formatted(context.getOperationMetadata().operationType(),
+				context.getOperationMetadata().provider());
+	}
+
+	@Override
+	public KeyValues getLowCardinalityKeyValues(ChatClientObservationContext context) {
+		return KeyValues.of(springAiKind(context), aiOperationType(context), aiProvider(context), stream(context));
+	}
+
+	@Override
+	public KeyValues getHighCardinalityKeyValues(ChatClientObservationContext context) {
+		return KeyValues.of(toolFunctionNames(context), toolFunctionCallbacks(context), chatClientAvisor(context),
+				chatClientAvisorParam(context));
+	}
+
+	protected KeyValue springAiKind(ChatClientObservationContext context) {
+		return KeyValue.of(ChatClientObservationDocumentation.LowCardinalityKeyNames.SPRING_AI_KIND, "chat.client");
+	}
+
+	protected KeyValue stream(ChatClientObservationContext context) {
+		return KeyValue.of(LowCardinalityKeyNames.STREAM, "" + context.isStream());
+	}
+
+	protected KeyValue aiOperationType(ChatClientObservationContext context) {
+		return KeyValue.of(ChatModelObservationDocumentation.LowCardinalityKeyNames.AI_OPERATION_TYPE,
+				context.getOperationMetadata().operationType());
+	}
+
+	protected KeyValue aiProvider(ChatClientObservationContext context) {
+		return KeyValue.of(ChatModelObservationDocumentation.LowCardinalityKeyNames.AI_PROVIDER,
+				context.getOperationMetadata().provider());
+	}
+
+	protected KeyValue toolFunctionNames(ChatClientObservationContext context) {
+		if (CollectionUtils.isEmpty(context.getRequest().getFunctionNames())) {
+			return CHAT_CLIENT_TOOL_FUNCTION_NAMES_NONE;
+		}
+		return KeyValue.of(ChatClientObservationDocumentation.HighCardinalityKeyNames.CHAT_CLIENT_TOOL_FUNCTION_NAMES,
+				context.getRequest().getFunctionNames().stream().collect(Collectors.joining(",")));
+	}
+
+	protected KeyValue toolFunctionCallbacks(ChatClientObservationContext context) {
+		if (CollectionUtils.isEmpty(context.getRequest().getFunctionCallbacks())) {
+			return CHAT_CLIENT_TOOL_FUNCTION_CALLBACKS_NONE;
+		}
+		return KeyValue.of(
+				ChatClientObservationDocumentation.HighCardinalityKeyNames.CHAT_CLIENT_TOOL_FUNCTION_CALLBACKS,
+				context.getRequest()
+					.getFunctionCallbacks()
+					.stream()
+					.map(fc -> fc.getName())
+					.collect(Collectors.joining(",")));
```

Comment:
For all values containing lists, I would recommend using the same format we used in the lower-level observations and put them between quotes, inside square brackets. That will make it easier to process list values consistently across different observations (for example, when drawing dashboards in Grafana or when reading the observability data via external integrations)

https://github.com/spring-projects/spring-ai/blob/main/spring-ai-core/src/main/java/org/springframework/ai/chat/observation/DefaultChatModelObservationConvention.java#L150